clang-tools-extra/clangd/Compiler.cpp

```diff
@@ -83,6 +83,7 @@ buildCompilerInvocation(const ParseInputs &Inputs, clang::DiagnosticConsumer &D,
   CI->getPreprocessorOpts().PCHThroughHeader.clear();
   CI->getPreprocessorOpts().PCHWithHdrStop = false;
   CI->getPreprocessorOpts().PCHWithHdrStopCreate = false;
+  CI->getPreprocessorOpts().SetUpStaticAnalyzer = true;
```

Comment:
Just so we are clear this might be an issue going forward, but without this ensure kokkos function doesn't work in clangd

Reply:
It possibly turns on code paths that wouldn't compile when you are actually running the compiler.  If downstream projects use `__clang_analyzer__` to do stuff for their projects, this might interfere with their usage of our clangd.  The alternative is that we define something like `__kokkos_analyzer__` that shouldn't have any conflicts, but I think this is less work until someone complains. 

I don't know why clangd doesn't turn this on in the upstream llvm, but I'm going to appeal to  to https://en.wikipedia.org/wiki/Wikipedia:Chesterton%27s_fence and not investigate farther.